### DISCUSSION TITLE
Space or location for quick ADF questions?

### DISCUSSION THREAD

**Discussion by cecilehannay** [NCAR/ADF#127] [General]
I am wondering if we should set a space somewhere for ADF questions.

Today, @richling@ucar.edu had a question and he posted it on the AMP@SE_help. He got a reply there. 
However, because there is not a complete overlap between the people in the AMP@SE_help, we may want to think about an ADF-specific location 

I wonder what the best place to host ADF quick questions would be. 
- We could create a space for ADF (something like AMP@ADF_help). 
But what I don't like about AMP@ADF_help is that it disappears after 24 hours. There is plenty of useful information/reply and it is basically lost after 24 hours. It is kind of a shame. 
There has been discussions about trying to convince UCAR that Google Space have retention longer than 24 hour but it is not clear whether/when it will happen. In the mean time, I think we need a location easily searchable and that doesn't disappear. 

- We could have a Zulip channel. I personally find that Zulip is not always easily searchable. But at least it doesn't disappear. 

- Would there be another option? Some where on github?    

I feel that we are having the discussion over and over. But I am still hopeful that one day, we come up with the perfect solution.

**Comment by andrewgettelman** [comment]
We are supposed to get the google spaces history thing changed. It does say 'history is on' but I don't see how to go backwards.

I can check with The CIO and Legal about it....

Regards,

Andrew

**Reply by cecilehannay** [reply]
'History on' still means '24 hour history'. It would be good to check with Legal when you can. 

To make it clear: are you recommending to create a AMP@ADF_help or to use AMP@SE_help? 

Another potential issue with Space is that we cannot add people outside of the organization. Maybe not an issue for now as all the developers are CGD.

**Comment by andrewgettelman** [comment]
I think a new channel/space for ADF is wise.

I assume that issue with spaces is also changable? (why cannot anyone with a google account use it)?

The lack of ability to have outside people I think argues for zulip (assuming we can get outside people on zulip).

But it doesn't solve the problem of spaces is what we are using and what is active and immediate for us (which would argue for spaces).

**Reply by cecilehannay** [reply]
> I assume that issue with spaces is also changable? (why cannot anyone with a google account use it)?

This is also a question for Legal.

**Comment by nusbaume** [comment]
I am happy with setting up our own AMP@ADF_help, especially because I am basically the only SE at the moment who is working on the ADF, so the other SEs might get annoyed if SE_help gets flooded with ADF-specific questions (as opposed to CAM or CESM-related questions).

Also, to be honest for ADF-specific questions I am not sure what Zulip buys you that isn't already provided by the Github Discussions page here.  In fact Github allows you to label a particular discussion thread as a "question", and then allows you to select the best answer, so that any future searches for a similar question will show that specific answer (similar to websites like Stack Overflow, if you have ever used that).  Of course, for general python-related questions the current Zulip forums would probably be the best choice.

Anyways, that's my two cents!

**Comment by andrewgettelman** [comment]
Let's try a space as Jesse suggests for immediate questions as came up yesterday. I agree though that Github discussions can probably be as useful as zulip. Good point Jesse. 

The advantage of spaces is that most people are on it during work hours, so responses are fast. 

Github discussions (here) can be more for things that need archiving (e.g., we should probably make sure our discussion of configurations yesterday ends up here or on the wiki).

**Comment by cecilehannay** [comment]
We started a Google Space called: 'AMP@ADF_help' that can be used for quick questions. 
If you want to get an invite (and you didn't get one already), please let Jesse (nusbaume@ucar.edu) or Cecile (hannay@ucar.edu) know.

**Comment by juliecaron** [comment]
i'd like an invite please

**Reply by cecilehannay** [reply]
Hi @juliecaron 

You were in the initial invite so you should have access to it already. On the left side column in your gmail, you should have access to Google Space. It should look something like:
 
![Screen Shot 2022-03-10 at 10 59 45 AM](https://user-images.githubusercontent.com/9723220/157726289-ae5d73a6-6a55-4331-abc3-17e0b4b214b9.png)

Maybe let me know if you cannot see it.

**Comment by andrewgettelman** [comment]
FYI: talked with Greg Madden (UCAR CIO): the ability to save google chat/spaces (history) should go live next week via a Sundog announcement....

**Comment by juliecaron** [comment]
i see it! thank you!
